Nidaulzafar node.js w2

NidaulZafar opened the pull request.

mnvacym commented on homework/hackyourtemperature/package.json:
```diff
@@ -0,0 +1,27 @@
+  "main": "server.js",
+  "type": "module",
+  "scripts": {
+    "test": "jest '__tests/app.test.js'",
```
Your test script is not working. By default Jest will already pick tests from `__tests__` folder. So you can only run jest here

NidaulZafar replied:
fixed.. thank you

mnvacym commented on homework/hackyourtemperature/__tests__/app.test.js:
```diff
@@ -0,0 +1,29 @@
+    // const data = JSON.parse(response.text);
+    expect(response.status).toEqual(404);
+  });
+  it("should return status 404 if no cityName provided", async () => {
```
just a small typo, it needs to be 400 I think :)

NidaulZafar replied:
yes, sorry..

mnvacym commented on homework/hackyourtemperature/app.js:
```diff
@@ -0,0 +1,29 @@
+  const cityName = req.body.cityName;
+  const resp = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&APPID=${API_KEY}`);
+  const data = await resp.json();
+  if (data.cod === String(404)) {
```
As an alternative, you could've written `resp.status === 404`. But this is also fine

mnvacym commented on homework/hackyourtemperature/app.js:
```diff
@@ -0,0 +1,29 @@
+  res.setHeader('Content-Type', 'application/json')
+  const cityName = req.body.cityName;
+  const resp = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cityName}&APPID=${API_KEY}`);
+  const data = await resp.json();
```
This is totally fine. Just fyi, you can destructure your code and make it more readable.
`const { cod, name, main } = await resp.json();`
And then, when you want to use variables, you can use them directly, and you don't need to do `data.`

NidaulZafar replied:
yes, it makes it simpler.. i have fixed it.. thank you 👍🏻

mnvacym reviewed (changes requested):
Hi Nida,

Well done! You did great! Just check your test script, it is currently broken.

Kind regards,
Ammar Merakli

NidaulZafar commented:
Hi.. thank you for your feedback.. I have implemented what you said. Only the test script in package.json file is confusing me. I have changed it to 
`"test": "jest"`
Is it correct?

mnvacym commented:
Yes it's correct
